x/validators/controller.go

```diff
@@ -0,0 +1,49 @@
+package validators
+
+import (
+	"github.com/confio/weave"
+	"github.com/confio/weave/errors"
+	"github.com/confio/weave/orm"
+	abci "github.com/tendermint/abci/types"
+)
+
+// Controller is the functionality needed by
+// cash.Handler and cash.Decorator. BaseController
+// should work plenty fine, but you can add other logic
+// if so desired
+type Controller interface {
+	CanUpdateValidators(store weave.KVStore, checkAddress CheckAddress, diff []abci.Validator) ([]abci.Validator, error)
+}
+
+// BaseController is a simple implementation of controller
+// wallet must return something that supports AsSet
+type BaseController struct {
+	bucket orm.Bucket
+}
+
+// NewController returns a basic controller implementation
+func NewController(bucket orm.Bucket) BaseController {
+	return BaseController{bucket: bucket}
+}
+
+func (c BaseController) CanUpdateValidators(store weave.KVStore, checkAddress CheckAddress, diff []abci.Validator) ([]abci.Validator, error) {
```

Comment:
pp: rename function to `UpdateValidators` please. it is doing something not answering a question.

Reply:
@alpe I renamed it because of @ethanfrey comment and it makes sense, because updatingValidators happens as part of abci spec.